### PR TITLE
better documentation for our input formats

### PR DESCRIPTION
**Pull request by Dieterbe**

### CODE REVIEW

File: docs/inputs.md [hunk 1]
```diff
@@ -23,16 +23,38 @@ note: it does not implement [carbon2.0](http://metrics20.org/implementations/)
...
+This format contains the data points as well as all metric identity data and metadata, in messagepack encoded messages (not JSON).
+This format is rather verbose and inefficient to encode/decode.
+See the [MetricData](https://godoc.org/github.com/raintank/schema#MetricData) documentation for format and implementation details.
+
+### MetricPoint
+
+This format is a hand-written binary format that is much more compact and fast to encode/decode compared to MetricData.
+See the [MetricPoint](https://godoc.org/github.com/raintank/schema#MetricPoint) documentation for format and implementation details.
+It is a minimal format that only contains the series identifier, value and timestamp.
+As such, it is paramount that MetricPoint messages for each series has been preceded by a MetricData message for that series.
```
Comment (woodsaj) [inline docs/inputs.md]: s/has been preceded/have been preceded/

Comment (woodsaj) [reply docs/inputs.md]: The true requirement is just that an entry exists in the index for the id of the metricPoint.

File: docs/inputs.md [hunk 2]
```diff
@@ -23,16 +23,38 @@ note: it does not implement [carbon2.0](http://metrics20.org/implementations/)
...
+Otherwise metrictank will not recognize the ID and will not be able handle it and discard the message.
```
Comment (woodsaj) [inline docs/inputs.md]: `Otherwise metrictank will not recognize the ID and the point will be discarded.`

File: docs/inputs.md [hunk 3]
```diff
@@ -46,8 +46,9 @@ See the [MetricData](https://godoc.org/github.com/raintank/schema#MetricData) do
 This format is a hand-written binary format that is much more compact and fast to encode/decode compared to MetricData.
 See the [MetricPoint](https://godoc.org/github.com/raintank/schema#MetricPoint) documentation for format and implementation details.
 It is a minimal format that only contains the series identifier, value and timestamp.
-As such, it is paramount that MetricPoint messages for each series has been preceded by a MetricData message for that series.
-Otherwise metrictank will not recognize the ID and will not be able handle it and discard the message.
+As such, it is paramount that MetricPoint messages for each series have been preceded by a MetricData message for that series, so
+that metrictank has had the change to add the metric information into its index.
```
Comment (replay) [inline docs/inputs.md]: `change` -> `chance`?

**Pull request merged**